app/templates/index.html

```diff
@@ -19,9 +19,19 @@
 
 <div id="preferences">
   <h3>Your Favourites</h3>
+  <form method="POST" action="{{ url_for('add_preference') }}">
+    {{ form.csrf_token }}
+    <div class="row">
+      {{ form.food.label }} {{ with_errors(form.food) }}
+    </div>
+    {{ form.submit(class_='button-primary') }}
+  </form>
   <ul>
     {% for pref in current_user.preferences %}
-    <li>{{ pref.food }}</li>
+    <li class="preference" data-id="{{ pref.id }}">
+      <span>{{ pref.food }}</span>
+      <a href="javascript:void(0)" class="delete">X</a>
```

Comment:
Boo!  Buttons for buttons.  Or clickable div with `role=button` and `tabindex` attributes.